src/handlers.ts

```diff
@@ -181,128 +184,159 @@ const ReplyActionSchema = z.object({
+            attributionActor = await lookupActor(
+                apCtx,
+                objectToReplyTo.attributionId.href,
+            );
+        }
 
-    if (!attributionActor) {
-        return new Response(null, {
-            status: 400,
+        if (!attributionActor) {
+            return new Response(null, {
+                status: 400,
+            });
+        }
+
+        const to = PUBLIC_COLLECTION;
+        const cc = [
+            attributionActor,
+            apCtx.getFollowersUri(ACTOR_DEFAULT_HANDLE),
+        ];
+
+        const conversation =
+            objectToReplyTo.replyTargetId || objectToReplyTo.id!;
+        const mentions = [
+            new Mention({
+                href: attributionActor.id,
+                name: attributionActor.name,
+            }),
+        ];
+
+        const replyId = apCtx.getObjectUri(Note, {
+            id: uuidv4(),
         });
-    }
 
-    const to = PUBLIC_COLLECTION;
-    const cc = [attributionActor, apCtx.getFollowersUri(ACTOR_DEFAULT_HANDLE)];
+        const reply = new Note({
+            id: replyId,
+            attribution: actor,
+            replyTarget: objectToReplyTo,
+            content: data.content,
+            summary: null,
+            published: Temporal.Now.instant(),
+            contexts: [conversation],
+            tags: mentions,
+            to: to,
+            ccs: cc,
+        });
 
-    const conversation = objectToReplyTo.replyTargetId || objectToReplyTo.id!;
-    const mentions = [
-        new Mention({
-            href: attributionActor.id,
-            name: attributionActor.name,
-        }),
-    ];
+        const createId = apCtx.getObjectUri(Create, {
+            id: uuidv4(),
+        });
 
-    const replyId = apCtx.getObjectUri(Note, {
-        id: uuidv4(),
-    });
+        const create = new Create({
+            id: createId,
+            actor: actor,
+            object: reply,
+            to: to,
+            ccs: cc,
+        });
 
-    const reply = new Note({
-        id: replyId,
-        attribution: actor,
-        replyTarget: objectToReplyTo,
-        content: data.content,
-        summary: null,
-        published: Temporal.Now.instant(),
-        contexts: [conversation],
-        tags: mentions,
-        to: to,
-        ccs: cc,
-    });
+        const activityJson = await create.toJsonLd();
 
-    const createId = apCtx.getObjectUri(Create, {
-        id: uuidv4(),
-    });
+        await ctx.get('globaldb').set([create.id!.href], activityJson);
+        await ctx.get('globaldb').set([reply.id!.href], await reply.toJsonLd());
 
-    const create = new Create({
-        id: createId,
-        actor: actor,
-        object: reply,
-        to: to,
-        ccs: cc,
-    });
+        await addToList(ctx.get('db'), ['outbox'], create.id!.href);
 
-    const activityJson = await create.toJsonLd();
+        const account = await accountService.getByApId(actor?.id!);
```

Comment:
Let's avoid using `?` and `!` here and instead check that these properties exist before hand - if they don't then we probably don't want to handle this.

Reply:
Although I just realised that `actor` here is referring to the actor of the user doing the reply - in this case we can use the `getDefaultAccount` method instead and forget about the `actor` completely